src/commands/txCommands/pullAll.js

```diff
@@ -0,0 +1,73 @@
+const chalk = require('chalk');
+const child_process = require('child_process');
+const Rx = require('rxjs');
+const txlib = require('./util');
+const txPullResource = require('./util/pullResourceTrns');
+
+const getProjectResourcesList$ =
+	txlib.resources$
+		.flatMap(Rx.Observable.from)
+		.reduce((resources, resource) => [ ...resources, resource ], []);
+
+/**
+ * recursively goes through list of resources and pulls trn content down
+ */
+const pullResource = (resources, iteration, commitOnComplete) => {
+	console.log(chalk.cyan(`Starting tx:pull for '${resources[iteration]}'`));
+	txPullResource.pullResourceContent$(resources[iteration]).subscribe(null, null, () => {
+		console.log(chalk.green(`Completed tx:pull for '${resources[iteration]}'`));
+
+		const doNext = iteration+1 < resources.length;
+
+		// only do a git commit if this is requested by the user when starting tx pull
+		if (commitOnComplete) {
+			// do an initial check to see if there is actually something to commit
+			child_process.exec('git status', (error, stdout, stderr) => {
+				// if 'nothing to commit' is NOT in `stdout` make a new commit
+				// with resource name in commit message
+				if (stdout.includes('nothing to commit') === false) {
+					child_process.execSync('git add src/trns');
+					// note: adjusting resource name to prevent branch/resources what contain
```

Comment:
typo: `what` -> `that`

alternative wording: "modify branch name to avoid triggering branch-aware Jira git hooks"